python/lib/sift_py/data_import/rosbags.py

```
@@ -0,0 +1,275 @@
+
+                # Flatten and collect all underlying fields in this message as RosChannels
+                msg_def = typestore.get_msgdef(connection.msgtype)
+                for field in msg_def.fields:
+                    key = get_key(connection, msg_def, field)
+                    if key in ros_channels:
+                        raise Exception(f"Duplicate key: {key}")
+                    ros_channels[key] = RosChannel.get_underlying_fields(
+                        sanitize(connection.topic), field, typestore
+                    )
+
+            headers = ["time"] + [
+                c.channel_name for channels in ros_channels.values() for c in channels
+            ]
+            w = csv.DictWriter(dst_file, headers)
+            w.writeheader()
+
+            print("Processing rosbag messages")
+            pbar = tqdm(total=reader.message_count, unit="messages")
+            for connection, timestamp, raw_data in reader.messages():
+                pbar.update(1)
+                if connection.msgtype not in registered_msg_types:
+                    if ignore_errors:
+                        continue
+                    else:
+                        raise Exception(
+                            f"Message type {connection.msgtype} not found in custom types."
+                        )
+
+                row: Dict[str, Union[int, float, bool, str]] = {}
+                msg = typestore.deserialize_cdr(raw_data, connection.msgtype)
+                msg_def = typestore.get_msgdef(connection.msgtype)
+                row["time"] = timestamp
+
+                if connection.topic in handlers:
+                    handlers[connection.topic](connection.topic, timestamp, msg)
+
+                for field in msg_def.fields:
+                    key = get_key(connection, msg_def, field)
+                    if key not in ros_channels:
+                        if ignore_errors:
+                            continue
+                        else:
```

tiniest nit, could just do `if not ignore errors: raise` to save a couple lines

---

We still need the other branch of the if statement to `continue` the loop.